### PR TITLE
Modules

### PR DESCRIPTION
**Pull request by moloshniy**
Added base Architecture

### CODE REVIEW

**Comment by moloshniy** [comment]
Added modules

File: app/build.gradle [hunk 1]
```diff
@@ -28,8 +28,4 @@ dependencies {
     implementation"org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
     implementation 'androidx.appcompat:appcompat:1.1.0'
```
Comment (gridasergienko) [inline app/build.gradle]: Please use const file for versions.

File: app/build.gradle [hunk 2]
```diff
@@ -28,8 +28,4 @@ dependencies {
...
     implementation 'androidx.core:core-ktx:1.2.0'
```
Comment (gridasergienko) [inline app/build.gradle]: Please use const file for versions.

File: build.gradle [hunk 3]
```diff
@@ -1,11 +1,14 @@
+apply from: 'buildSystem/dependencies.gradle'
+apply from: 'buildSystem/const.gradle'
+
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 
 buildscript {
     ext.kotlin_version = '1.3.61'
```
Comment (gridasergienko) [inline build.gradle]: You have different version in `dependencies.gradle`

File: common/data/build.gradle [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            proguardFiles getDefaultProguardFile(defaultProguardFile), proguardRules
+        }
+    }
+}
+
+dependencies {
+    api fileTree(dir: 'libs', include: ['*.jar'])
+    api rootProject.ext.kotlinDependencies.values()
+    testImplementation rootProject.ext.testJunitDependencies.values()
+    androidTestImplementation rootProject.ext.testAndroidJunitDependencies.values()
```
Comment (gridasergienko) [inline common/data/build.gradle]: not sure that we need android testing for data modules. Suggest removing it from data and leave only in ui.

File: common/data/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<resources>
+    <string name="app_name">com-data</string>
```
Comment (gridasergienko) [inline common/data/src/main/res/values/strings.xml]: We don't really need it, please clean it up.

File: common/domain/src/main/java/com/example/barcodeapp/com/domain/ComDomain.java [hunk 6]
```diff
@@ -0,0 +1,4 @@
+package com.example.barcodeapp.com.domain;
+
+public class ComDomain {
```
Comment (gridasergienko) [inline common/domain/src/main/java/com/example/barcodeapp/com/domain/ComDomain.java]: Is this class already used somewhere, if not Please remove this class and fix the package name. com....com... is inconvenient.

File: common/nav/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<resources>
+    <string name="app_name">com-nav</string>
```
Comment (gridasergienko) [inline common/nav/src/main/res/values/strings.xml]: Useless var

File: common/ui/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+<resources>
+    <string name="app_name">com-ui</string>
```
Comment (gridasergienko) [inline common/ui/src/main/res/values/strings.xml]: useless string

**Pull request closed without merging**